### PR TITLE
Add some helper methods for working with ServerBossBars

### PR DESCRIPTION
**Pull request by kashike**
[**API**](https://github.com/SpongePowered/SpongeAPI/pull/1534) | [Common](https://github.com/SpongePowered/SpongeCommon/pull/1296)
Adds #1533

Test plugin: https://gist.github.com/kashike/bb89aee16738a68f078a6e0c82432583

### CODE REVIEW

File: src/main/java/org/spongepowered/api/entity/living/player/Player.java [hunk 1]
```diff
@@ -339,4 +343,42 @@ default GameModeData getGameModeData() {
...
+     *
+     * @param bar The boss bar
+     * @see ServerBossBar#removePlayer(Player)
+     */
+    default void removeBossBar(final ServerBossBar bar) {
+        checkNotNull(bar, "bar").removePlayer(this);
+    }
+
+    /**
+     * Gets an unmodifiable collection of boss bars this player is currently in.
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/api/entity/living/player/Player.java]: While technically correct, this sounds odd.

**Review by ST-DDT — APPROVED** [review]
I like the last method. The other ones just look like inversion of controll, but for usability reasons i guess its ok.

File: src/main/java/org/spongepowered/api/entity/living/player/Player.java (hunk 1 above)
Comment (kashike) [reply src/main/java/org/spongepowered/api/entity/living/player/Player.java]: What would you recommend?

Comment (ST-DDT) [reply src/main/java/org/spongepowered/api/entity/living/player/Player.java]: `... that are shown to this player` or `... that apply to this player`. Sorry i should have proposed these directly, but I'm not sure about them either.

Comment (parlough) [reply src/main/java/org/spongepowered/api/entity/living/player/Player.java]: `Gets an unmodifiable collection of the boss bars this player is currently viewing.`

File: src/main/java/org/spongepowered/api/entity/living/player/Player.java [hunk 2]
```diff
@@ -339,4 +343,42 @@ default GameModeData getGameModeData() {
...
+     *
+     * @param bar The boss bar
+     * @see ServerBossBar#addPlayer(Player)
+     */
+    default void addBossBar(final ServerBossBar bar) {
+        checkNotNull(bar, "bar").addPlayer(this);
+    }
+
+    /**
+     * Removes this player to the boss bar.
```
Comment (JonathanBrouwer) [inline src/main/java/org/spongepowered/api/entity/living/player/Player.java]: Is this correct english?
I think it should be 'Removes this player from the boss bar.'

File: src/main/java/org/spongepowered/api/entity/living/player/Player.java [hunk 3]
```diff
@@ -339,4 +343,42 @@ default GameModeData getGameModeData() {
...
+     * @see ServerBossBar#removePlayer(Player)
+     */
+    default void removeBossBar(final ServerBossBar bar) {
+        checkNotNull(bar, "bar").removePlayer(this);
+    }
+
+    /**
+     * Gets an unmodifiable collection of boss bars this player is currently in.
+     *
+     * @return A collection of boss bars this player is currently in
```
Comment (parlough) [inline src/main/java/org/spongepowered/api/entity/living/player/Player.java]: Change this to match whatever you choose for ST-DDT's suggestion.

**Review by parlough — APPROVED** [review]
I think this is a good addition and simply makes sense to have.

**Comment by Zidane** [comment]
@kashike can you update the comments above for the suggestions so I can look into merging this.

**Comment by ryantheleach** [comment]
Talked to kashike, Current status is a slight rework in the relationship between players and bossbars vs their helper methods. I don't know exact details, but have heard there will be work on this soon.

**Comment by gabizou** [comment]
@kashike @Zidane would we want this in API 7 at all?

**Pull request merged**